dagger-reflect/reflect/src/main/java/dagger/reflect/ReflectiveMembersInjector.java

```diff
@@ -75,29 +78,37 @@
 
         methodBindings.put(method, bindings);
       }
+      hierarchyBindings.add(new Bindings(fieldBindings, methodBindings));
 
       target = target.getSuperclass();
     }
+    // [@Inject] Fields and methods in superclasses are injected before those in subclasses.
+    // we were walking up (getSuperclass), but spec says @Inject should be walking down the hierarchy
+    Collections.reverse(hierarchyBindings);
```

Comment:
The complexity of this will be crazy because it does random access on the LinkedList. Just add items to the head of the list instead of the tail.

Reply:
Doc says it runs in linear: https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#reverse(java.util.List)

LinkedList is not RandomAccess, so a pair of ListIterators will be used.

Reply:
I still think it's worth changing. You can use `addFirst` to self-document the intent a bit.